dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/TrackerPreheat.java

```diff
@@ -643,11 +645,10 @@ public Relationship getRelationship( org.hisp.dhis.tracker.domain.Relationship r
 
     public boolean isDuplicate( org.hisp.dhis.tracker.domain.Relationship relationship )
     {
-        RelationshipType relationshipType = get( RelationshipType.class, relationship.getRelationshipType() );
-
         if ( hasRelationshipKey( relationship ) )
         {
-            RelationshipKey relationshipKey = getRelationshipKey( relationship );
+            RelationshipType relationshipType = get( RelationshipType.class, relationship.getRelationshipType() );
+            RelationshipKey relationshipKey = getRelationshipKey( relationship, relationshipType.getUid() );
```

Comment:
do we always call `relationshipType.getUid()` when calling `getRelationshipKey()`?

if so we should move that `getUid()` inside of `getRelationshipKey()`
I was just reminded because in the PR description you said we should make sure to always store the UID of the relationship key. This way we make sure we do.